packages/network-controller/src/NetworkController.ts

```diff
@@ -658,6 +679,18 @@ export class NetworkController extends BaseControllerV2<
       delete state.networkConfigurations[networkConfigurationId];
     });
   }
+
+  /**
+   * Rolls back provider config to the previous provider in case of errors or inability to connect during network switch.
+   */
+  rollbackToPreviousProvider() {
+    const { type, id } = this.previousProvider;
+    if (type === RPC && id) {
+      this.setActiveNetwork(id);
+    } else {
+      this.setProviderType(type);
+    }
```

Comment:
Following the above suggestion this could be:

``` suggestion
    const specifier = this.#previousNetworkSpecifier;
    if (Object.values(NetworkType).includes(specifier)) {
      this.setProviderType(type);
    } else {
      this.setActiveNetwork(id);
    }
```

I guess in that case `NetworkType` would need to be an enum instead of just a type, as it is now. Though that might provide useful in tests.

Reply:
done here: https://github.com/MetaMask/core/pull/1132/commits/5fb32845402506720c09adebda05989ffa86e791